eng/common/TestResources/Remove-TestResources.ps1

```diff
@@ -137,9 +137,31 @@ if (![string]::IsNullOrWhiteSpace($ServiceDirectory)) {
     }
 }
 
+$verifyDeleteScript = {
+    try {
+        $group = Get-AzResourceGroup -name $ResourceGroupName
+    } catch {
+        if ($Error[0].ToString().Contains("Provided resource group does not exist")) {
+            Write-Verbose "Resource group '$ResourceGroupName' not found. Continuing..."
+            return
+        }
+        throw $Error[0]
+    }
+
+    if ($group.ProvisioningState -ne "Deleting")
+    {
+        throw "Resource group is in '$($group.ProvisioningState)' state, expected 'Deleting'"
+    }
+}
+
 Log "Deleting resource group '$ResourceGroupName'"
-if (Retry { Remove-AzResourceGroup -Name "$ResourceGroupName" -Force:$Force }) {
-    Write-Verbose "Successfully deleted resource group '$ResourceGroupName'"
+if ($Force) {
+    Remove-AzResourceGroup -Name "$ResourceGroupName" -Force:$Force -AsJob
+    Retry $verifyDeleteScript 3
```

Comment:
You probably should log that you scheduled a job (and I'd used the term "job" as well, being PowerShell-idiomatic), but is there much point to verifying? You could print the job ID and if people cared to see output, could use `get-job` to view it's status. If the goal is just to fire and forget, verifying seems unnecessary.

Reply:
I think the issue @heaths is we have a degree of uncertainty about how reliably the "fire and forget, then exit the script" will succeed in making the call to ARM before the shell is killed. Originally I had it as solely fire and forget but added verify after review comments.